### DISCUSSION TITLE
wsl-win10 ssh config synchronization

### DISCUSSION THREAD

**Discussion by CptnGreen** [microsoft/WSL#6522] [General] [answered]
**Is your feature request related to a problem? Please describe.**
It would be great to have an ability to easily synchronize  Windows (`C:\Users\<username>\.ssh`) and WSL (`/home/<username>/.ssh`) `.ssh` folders (config and keys).

**Describe the solution you'd like**
Natural solution would be to symlink one folder to another but this approach fails because of the permissions' differences (WSL treats windows' files permissions as `777`).

**Describe alternatives you've considered**
The heavier solution would be to implement some script which performs synchronization after each modification in either folder.

**Additional context**
Sorry if I duplicated another issue - I looked for it but couldn't find one (promise).

**Comment by DHowett** [comment, marked as answer]
If you mount /mnt/c with metadata=on you can set permissions on `config` that WSL will remember/respect. Would that help?

**Comment by therealkenc** [comment]
Moving over to discussion because any mechanism to sync would be a userspace project (WSL proper doesn't know `ssh` from any other program in Linux-land). Some `bash` or `python` script in a github. There is a StackExchange thread for Dustin's suggestion [here](https://superuser.com/questions/1183176/can-i-share-my-ssh-keys-between-wsl-and-windows).
